### ISSUE TITLE
Recovering the state of old Bioconductor releases 

### ISSUE THREAD

**Issue by LTLA** [Bioconductor/BiocManager#86]
When running `BiocManager::install()` with an older `version=`, the old Bioconductor packages are correctly pulled; however, the latest version of CRAN packages are also pulled. There is no guarantee that CRAN packages are back-compatible and this has led to a few problems (e.g., LTLA/BiocNeighbors#17). 

Can `install()` be smart enough to pull from archived versions of CRAN packages at the time of the last successful BBS build on the specified Bioconductor version? It seems like this should be possible with the **checkpoint** package, or Bioconductor could just construct an index of versions of available CRAN packages at the end of each release. This would avoid bugs caused by backward incompatibilities and facilitate recovery of the correct set of packages for reproducing old results.

**Comment by mtmorgan** [comment]
This is the topic of this branch https://github.com/Bioconductor/BiocManager/tree/add_mran which I am sitting on. It pulls from MRAN based on release dates in https://bioconductor.org/config.yaml . An work-around until that is available is

```
options(
    repos = c(CRAN = "https://cran.microsoft.com/snapshot/<SNAPSHOT DATE from https://cran.microsoft.com/snapshot/>")
)
```

and interesting approach would be to take the snapshot of pre-built binaries from RStudio

```
options(repos = c(CRAN = "<value from https://packagemanager.rstudio.com/client/#/repos/1/overview>"))
```
and apparently another very interesting workaround would be to get the Bioc packages from the same location
```
options(
    BioC_mirror = "https://packagemanager.rstudio.com/bioconductor",
    repos = c(CRAN = "<value from https://packagemanager.rstudio.com/client/#/repos/1/overview>")
)
```
All of these should work with `BiocManager::repositories()` and hence `BiocManager::install()`.

It would be great to get real-world experience with any of these approaches.

**Comment by LTLA** [comment]
The MRAN `repos` doesn't work because `BiocManager:::.repositories_base` replaces any `/snapshot/`-patterned URL with Rstudio's repo. The package manager works fine, though.

**Comment by hpages** [comment]
Also worth mentioning that this is already being discussed in issue #71.

**Comment by LiNk-NY** [comment]
This has been implemented here ab16fc2f19bc8a50084e369b85dd3f72682bb9a9.
